src/thor/bidirectional_astar.cc

```diff
@@ -1138,6 +1149,14 @@ std::vector<std::vector<PathInfo>> BidirectionalAStar::FormPath(GraphReader& gra
   return paths;
 }
 
+void BidirectionalAStar::ModifyHierarchyLimits() {
+  // Distance threshold optimized for unidirectional search. For bidirectional case
+  // they can be lowered.
+  // Decrease distance thresholds only for arterial roads for now
+  hierarchy_limits_forward_[1].expansion_within_dist /= 5.f;
+  hierarchy_limits_reverse_[1].expansion_within_dist /= 5.f;
```

Comment:
current default hierarchy limit for arterial roads is 100km; it's very costly for performance. This code decreases it to 20km making performance drop more or less reasonable

Reply:
i guess we have to do a big round of RAD to see what kind of changes we get though. I could imagine many routes will change because of this especially in rural areas where now you have to find an upward transition within 20km.

with respect to the performance difference how many routes are run with a retry and relaxed limits vs master. i wonder if we would have more retries with this change

Reply:
> i guess we have to do a big round of RAD to see what kind of changes we get though. I could imagine many routes will change because of this especially in rural areas where now you have to find an upward transition within 20km.

agree about RAD. I'm working on it right now

> with respect to the performance difference how many routes are run with a retry and relaxed limits vs master? i wonder if we would have more retries with this change

interesting note. I will check it

Reply:
@kevinkreiser about retries: it's approximately the same: 296 (master) vs 292 (branch)